### PR TITLE
improve error message in tests

### PR DESCRIPTION
**Pull request by hamelsmu**
This is how an error message is displayed after this improvement

```
>$ nbprocess_test --fname nbs/13_lookup.ipynb  
WARNING:root:TypeError in /Users/hamel/github/nbprocess/nbs/03_process.ipynb:
===========================================================================

While Executing Cell #6:
--------------------------------------------------
     1 def boom(): pass
     2
     3 #this is a comment.
     4 for i in range(10):
---> 5     boom(1,3)
     6
     7 some_other_code = 4
--------------------------------------------------
  File "<kernel-3-7b77017963ed>", line 5, in <module>
    boom(1,3)

TypeError: boom() takes 0 positional arguments but 2 were given



nbprocess Tests Failed On The Following Notebooks:
==================================================
	03_process.ipynb%
```

This is how errors appeared before this change, you have to wade through a big stack trace

image

### CODE REVIEW

File: nbprocess/test.py [hunk 1]
```diff
@@ -21,28 +21,42 @@ def _do_eval(cell, flags):
     if direc.get('eval:', [''])[0].lower() == 'false': return False
     return not flags & direc.keys()
 
+# %% ../nbs/14_test.ipynb 5
+def _format_code(code_list, lineno):
+    _fence = '-'*50
+    l=[_fence]
+    for i,c in enumerate(code_list, start=1):
```
Comment (jph00) [inline nbprocess/test.py:28]: Could this be done in 1 line with a list comprehension?

File: nbprocess/test.py [hunk 2]
```diff
@@ -21,28 +21,42 @@ def _do_eval(cell, flags):
...
     "Execute tests in notebook in `fn` except those with `skip_flags`"
     if not IN_NOTEBOOK: os.environ["IN_TEST"] = '1'
     flags=set(L(skip_flags)) - set(L(force_flags))
     k,start = NBRunner(),time.time()
     
     def _exec_cell(cell):
-        if _do_eval(cell, flags): k.run(cell)
-
-    if do_print: print(f'Starting {fn}')
+        try:
```
Comment (jph00) [inline nbprocess/test.py:42]: should the `try` and `if` be switched, so the `do_eval` isn't in `try`?

File: nbprocess/test.py [hunk 3]
```diff
@@ -21,28 +21,42 @@ def _do_eval(cell, flags):
...
+            if _do_eval(cell, flags): k.run(cell)
+        except Exception as e:
+            _fence = '='*75
+            line_no = e.__traceback__.tb_next.tb_next.tb_next.tb_next.tb_lineno
+            tb_str = '\n'.join(traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)[-2:])
+            cell_str = f"\nWhile Executing Cell #{cell.idx_}:\n{_format_code(cell.source.splitlines(), line_no)}"
+            warning(f"{type(e).__name__} in {fn}:\n{_fence}\n{cell_str}\n{tb_str}\n") 
+            raise Exception('nbprocess test failed')
```
Comment (jph00) [inline nbprocess/test.py:50]: we should probably raise a more specific subclass of `Exception`

File: nbprocess/test.py [hunk 4]
```diff
@@ -21,28 +21,42 @@ def _do_eval(cell, flags):
...
+        if do_print: print(f'Starting {fn}')
         with open(os.devnull, "w") as f, contextlib.redirect_stdout(f): NBProcessor(fn, _exec_cell).process()
         if do_print: print(f'- Completed {fn}')
         return True,time.time()-start
-    except Exception as e:
-        fence = '='*50
-        msg = f'{e}\n{fence}' if str(e) else ''
-        warn(f'\n\nError: {type(e).__name__} in {fn}:\n{fence}\n{msg}')
-        traceback.print_exc()
+    except:
```
Comment (jph00) [inline nbprocess/test.py:56]: ...then catch the subclass here?

**Review by jph00 — CHANGES_REQUESTED** [review]
This looks great! Just some minor suggestions

File: nbprocess/test.py (hunk 4 above)
Comment (hamelsmu) [reply nbprocess/test.py]: ok fixed this

File: nbprocess/test.py (hunk 3 above)
Comment (hamelsmu) [reply nbprocess/test.py]: Fixed this

**Pull request merged**